src/upgrade/upgrade_scripts/5.12.0/remove_allowed_buckets.js

```diff
@@ -0,0 +1,30 @@
+/* Copyright (C) 2022 NooBaa */
+"use strict";
+
+async function run({ dbg, system_store }) {
+    try {
+        dbg.log0('starting upgrade accounts...');
+        const accounts = system_store.data.accounts
```

Comment:
Is that what we want to do with upgrade? will it set that all the users have full control or no control? it seems like we need to handle such upgrades with care and at least let the users know what will be the artifacts.

Reply:
Bucket owners will still have full control over the buckets but yes, anyone relying on the `permission_list` or `full_permission` will face issues. Would you recommend generating S3 policies for them in the upgrade process?

Also, I don't think we can truly replicate `full_permission` because of the way we handle setting policies on the buckets.

Reply:
I think anyway most users are not using allowed_buckets (as we don't have any UI anymore). but I think users that didn't play with allowed_buckets and could access all of their buckets need to still be able to do that. was the default for newly created bucket is S3 full permission?

Reply:
Under the assumption that we are talking about users using NooBaa core in conjunction with NooBaa operator. 

When users create an OBC to get a bucket, we create it using operator's account (which has `full_permission: true`), so the bucket owner is the `operator`. Later on `operator` creates a new account which grants access to the bucket via `permission_list`.  So the `full_permission` is `true` for the account that creates the bucket but not for the account that is given to the user for accessing the bucket.

To address that, if in https://github.com/noobaa/noobaa-operator/pull/956 we replace [PermissionList](https://github.com/utkarsh-pro/noobaa-operator/blob/69896b9bd268bbac8fcf695adfb3a1633b5b636a/pkg/obc/provisioner.go#L600) with a S3 policy which allows access to the bucket, then the user should be able keep their access. But this approach has the following issues:
1. As you mentioned earlier, we would be creating policy without user's permission.
2. We will be adding policy to the bucket, which is fine but if the user would try to add some policy to the bucket (unaware of the previous policy set by the operator), they will overwrite the policy (which will block their own access). I am not sure how prevalent is this flow though where user set S3 policies on the OBC based buckets. 
    - We can kind of fix this though, by denying user (again via policy) to set any S3 policies on the bucket (again not sure how prevalent is this flow).

Reply:
I think that now the user creates this account and give if FC using bucket_claim_owner. so not sure we need any change to the noobaa-operator in that case.
but other than that, I think that we should decide on one approach - I think maybe we shouldn't add any other bucket policies but maybe worth discussing with more people. But we should let the user know that bucket-owner/system-owner will need to recreate all the removed access with new bucket policies.